config_system/general.py

```diff
@@ -769,26 +770,43 @@ def needs_inputbox(self):
             config = get_config(self.value)
             return config['datatype'] in ["string","int","hex"]
         return False
-    def set_bool(self, new_value):
+    def set(self):
+        """Sets a boolean option to true"""
+        if self.type in ["config", "menuconfig"]:
+            config = get_config(self.value)
+            if len(config['selected_by']) > 0:
+                # menuconfig shouldn't mark set by user if option can't be changed
+                return
+
+            if config['datatype'] == 'bool':
+                set_config(self.value, 'y')
+    def clear(self):
+        """Sets a boolean option to false"""
         if self.type in ["config", "menuconfig"]:
             config = get_config(self.value)
-            if len(config['selected_by']) > 0:  # menuconfig shouldn't mark set by user if option can't be changed
+            if len(config['selected_by']) > 0:
+                # menuconfig shouldn't mark set by user if option can't be changed
                 return
 
             if config['datatype'] == 'bool':
-                set_config(self.value, new_value)
+                set_config(self.value, 'n')
     def toggle(self):
         if self.type in ["config","menuconfig"]:
             config = get_config(self.value)
             if config['datatype'] != 'bool':
                 pass # Ignore
             elif config['value'] == 'y':
-                self.set_bool('n')
+                self.clear()
             else:
-                self.set_bool('y')
+                self.set()
     def get_value(self):
-        return get_config(self.value)['value']
+        """Always return a string representation"""
+        value = get_config(self.value)['value']
+        if get_config(self.value)['datatype'] == 'int':
+            value = str(value)
+        return value
     def set_value(self, new_value):
```

Comment:
What's stopping someone calling this on an int option?

There's also no set_int function - should there be? Otherwise it seems like you have to use set_value for both? (which would be incorrect, because it would store the menuconfig input text as a string, not an int)

Apologies if I've misunderstood :)

Reply:
That's right, get_value and set_value are intended to work with any type (int, bool, string) of option.
get_value is generally called to get the text for an input box. And set_value would be called to set the value after the inputbox is closed. Since an inputbox is not used to set the value on booleans, they should only see an int or string in this use case. The other usage is for some logging a little below...